### PR TITLE
[WebGPU] Push constants Storage i32 -> Uniform vector<4xi32>.

### PR DESCRIPTION
**Pull request by ScottTodd**
We discussed this [here on Discord](https://discord.com/channels/689900678990135345/867052513935753277/1047198677514600518).

When using `tensor<Nxi32>` we get an error like this from Tint:
```
Tint reported 1 error(s) for a SPIR-V program, see diagnostics:
error: line:36: Structure id 8 decorated as Block for variable in
  Uniform storage class must follow relaxed uniform buffer layout rules:
  member 0 contains an array with stride 4 not satisfying alignment to 16
```

Following the spec: https://www.w3.org/TR/WGSL/#address-space-layout-constraints, we should be aligning to 16. Switching to `tensor<Nxvector<4xi32>>` gives us that alignment.

This change packs a conceptual array of `i32`s into an array of `vector<4xi32>`s - possibly with unused elements at the end. Unpacking changes from using just `tensor::ExtractOp` to using `tensor::ExtractOp` (to get one of the vec4s) -> `vector::ExtractElementOp` (to get the specific element).

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/WGSL/WGSLReplacePushConstants.cpp [hunk 1]
```diff
@@ -124,13 +133,18 @@ class WGSLReplacePushConstantsPass
       alignmentAttr = constantLoadOps[0].getAlignmentAttr();
     }
 
+    // We could store into a tensor<Nxi32>, but vec4s are better supported, so
+    // we'll use tensor<Nxvector<4xi32>> instead.
+    uint64_t numberOfVec4s = maxConstantIndex / 4 + 1;
```
Comment (antiagainst) [inline compiler/src/iree/compiler/Codegen/WGSL/WGSLReplacePushConstants.cpp:138]: llvm has `llvm::divideCeil` for it. So if it's a perfect multiple of 4, we can avoid the `+1` there.

**Review by antiagainst — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/WGSL/WGSLReplacePushConstants.cpp (hunk 1 above)
Comment (ScottTodd) [reply compiler/src/iree/compiler/Codegen/WGSL/WGSLReplacePushConstants.cpp:138]: Oh, did I make an off by one error? Hmm.. I think here if the "max index" is 4, that means I'd need [0, 1, 2, 3, 4], which is two vec4s?

Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/WGSL/WGSLReplacePushConstants.cpp:138]: Ah you are right. This is index. Sorry I read too fast. :)

Comment (ScottTodd) [reply compiler/src/iree/compiler/Codegen/WGSL/WGSLReplacePushConstants.cpp:138]: No problem. I can add a comment too.

**Pull request merged**